tests/unit_tests.cpp

```diff
@@ -523,4 +545,248 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(id_api, _dimensions, test_dimensions::type) {
   // TODO: In-place and binary operators
 }
 
+// Helper type to construct unique kernel names for all instantiations of
+// a templated test case.
+template<typename T, int dimensions>
+struct kernel_name {};
```

Comment:
Is it really necessary to have a unique kernel name? hipSYCL is more permissive than the SYCL spec because CUDA/HIP don't require this. kernel invocations in hipSYCL *should* work without specifying a kernel name.

Reply:
It's certainly not *necessary*, but I did this so that 1) the tests are written in valid SYCL and thus 2) can be compiled with ComputeCpp or any other spec conformant SYCL implementation! I thought that might prove useful at some point (e.g. to check for differences between implementations), but of course I can take it out, if you prefer.

Reply:
No, it's fine, I was only interested in whether there were problems compiling without a unique kernel name, since that should work :)